CMakeLists.txt

```diff
@@ -146,7 +146,7 @@ add_custom_target(coverage_no_tests
 # CMake is not clever enough to build the tests before running them so we use
 # the 'check' target below that does both.
 add_custom_target(check
-  COMMAND ${CMAKE_CTEST_COMMAND}
+  COMMAND ${CMAKE_CTEST_COMMAND} --output-on-failure
```

Comment:
Ah! That was easier than expected. This means that running the executable directly will still output everything, including trace messages, but we can probably get around that by running the right ctest command.